contracts/argent_account.cairo

```diff
@@ -42,4 +44,47 @@ mod ArgentAccount {
     fn supportsInterface(interface_id: felt) -> bool {
         interface_id == ERC165_IERC165_INTERFACE_ID | interface_id == ERC165_ACCOUNT_INTERFACE_ID | interface_id == ERC165_OLD_ACCOUNT_INTERFACE_ID
     }
+
+    // ERC1271
+    #[view]
+    fn isValidSignature(ref signatures: Array::<felt>, hash: felt) -> bool {
+        let is_valid_signer = is_valid_signer_signature(ref signatures, hash);
+        let is_valid_guardian = is_valid_guardian_signature(ref signatures, hash);
+        is_valid_signer & is_valid_guardian
+    }
+
+    fn is_valid_signer_signature(ref signatures: Array::<felt>, hash: felt) -> bool {
+        assert(signatures.len() >= 2_u128, 'argent: signature format invalid');
+        let signature_r = signatures.at(0_u128);
+        let signature_s = signatures.at(1_u128);
+        ecdsa::check_ecdsa_signature(hash, signer::read(), signature_r, signature_s)
+    }
+
+    fn is_valid_guardian_signature(ref signatures: Array::<felt>, hash: felt) -> bool {
+        let guardian_ = guardian::read();
+        if guardian_ == 0 {
+            assert(signatures.len() == 2_u128, 'argent: signature format invalid');
+            return true;
+        }
+        assert(signatures.len() == 4_u128, 'argent: signature format invalid');
+        let signature_r = signatures.at(2_u128);
+        let signature_s = signatures.at(3_u128);
+        let is_valid_guardian_signature = ecdsa::check_ecdsa_signature(
+            hash, guardian_, signature_r, signature_s
+        );
+        if is_valid_guardian_signature {
+            return true;
+        }
+        ecdsa::check_ecdsa_signature(hash, guardian_backup::read(), signature_r, signature_s)
+    }
+
+    fn validate_signer_signature(ref signatures: Array::<felt>, hash: felt) {
```

Comment:
I think we can probably remove `validate_signer_signature ` and `validate_guardian_signature` and just inline the `assert` in the code if needed.

Reply:
thought the same actually

Reply:
removed